extism.go

```diff
@@ -34,10 +40,12 @@ type Runtime struct {
 
 // PluginConfig contains configuration options for the Extism plugin.
 type PluginConfig struct {
-	ModuleConfig  wazero.ModuleConfig
-	RuntimeConfig wazero.RuntimeConfig
-	EnableWasi    bool
-	LogLevel      LogLevel
+	ModuleConfig   wazero.ModuleConfig
+	RuntimeConfig  wazero.RuntimeConfig
+	EnableWasi     bool
+	LogLevel       LogLevel
+	ObserveAdapter *observe.AdapterBase
```

Comment:
`AdapterBase` and `Adapter` being two separate types is a bit problematic here, we need the `AdapterBase` because of `NewTraceContext` and It's nice to have `Adapter` because it has `Start`. Maybe `Adapter` should have an function that returns the `AdapterBase` so that we can have the best of both worlds?

Reply:
Since we are starting/stopping the adapter, `AdapterBase` is good enough